src/benchmarklib/benchmark_runner.cpp

```diff
@@ -26,7 +26,22 @@
 
 namespace opossum {
 
-BenchmarkRunner::BenchmarkRunner(const BenchmarkConfig& config,
+void BenchmarkRunner::reset(std::shared_ptr<BenchmarkConfig> config,
+                            std::unique_ptr<AbstractBenchmarkItemRunner> benchmark_item_runner,
+                            std::unique_ptr<AbstractTableGenerator> table_generator, const nlohmann::json& context) {
+  get() = BenchmarkRunner(config, std::move(benchmark_item_runner), std::move(table_generator), context);
+}
+
+BenchmarkRunner& BenchmarkRunner::operator=(BenchmarkRunner&& other) {
+  _config = std::move(other._config);
+  _benchmark_item_runner = std::move(other._benchmark_item_runner);
+  _table_generator = std::move(other._table_generator);
+  _results = std::move(other._results);
+  _context = std::move(other._context);
+  return *this;
+}
```

Comment:
Not sure why I can't use the _default_'ed move-assignment operator here instead of providing one. The compiler then always chooses the _delete_'d copy-assignment operator.